release/instance.go

```diff
@@ -0,0 +1,105 @@
+package release
+
+import (
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/instance"
+	"github.com/weaveworks/flux/platform"
+)
+
+// Operations on instances (or instance.* types) that we need for
+// releasing
+
+func LockedServices(config instance.Config) flux.ServiceIDSet {
+	ids := []flux.ServiceID{}
+	for id, s := range config.Services {
+		if s.Locked {
+			ids = append(ids, id)
+		}
+	}
+	idSet := flux.ServiceIDSet{}
+	idSet.Add(ids)
+	return idSet
+}
+
+// CollectAvailableImages is a convenient shim to
+// `instance.CollectAvailableImages`.
+func CollectAvailableImages(inst *instance.Instance, updateable []*ServiceUpdate) (instance.ImageMap, error) {
+	var servicesToCheck []platform.Service
+	for _, update := range updateable {
+		servicesToCheck = append(servicesToCheck, update.Service)
+	}
+	return inst.CollectAvailableImages(servicesToCheck)
+}
+
+// applyChanges effects the calculated changes on the platform.
+func applyChanges(inst *instance.Instance, updates []*ServiceUpdate, results flux.ReleaseResult) error {
+	// Collect definitions for each service release.
+	var defs []platform.ServiceDefinition
+	// If we're regrading our own image, we want to do that
+	// last, and "asynchronously" (meaning we probably won't
+	// see the reply).
+	var asyncDefs []platform.ServiceDefinition
+
+	for _, update := range updates {
+		namespace, serviceName := update.ServiceID.Components()
+		switch serviceName {
+		case FluxServiceName, FluxDaemonName:
+			inst.LogEvent(namespace, serviceName, "Starting. (no result expected)")
+			asyncDefs = append(asyncDefs, platform.ServiceDefinition{
+				ServiceID:     update.ServiceID,
+				NewDefinition: update.ManifestBytes,
+				Async:         true,
+			})
+		default:
+			inst.LogEvent(namespace, serviceName, "Starting")
```

Comment:
Starting what? This makes the history a bit weird. The only thing the history ever says now is: `helloworld: Starting`